tests/formats/format_helpers.test.ts

```diff
@@ -317,18 +339,44 @@ describe("formatValue on number", () => {
     );
   });
 
-  test("with brackets inside the string", () => {
-    expect(() => formatValue(1234, { format: "[$[]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$]]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$[]]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$][]#,##0.0", locale })).toThrow();
+  test("Can escape strings or characters in format", () => {
+    expect(formatValue(5, { format: "0\\%", locale })).toBe("5%");
+    expect(formatValue(5, { format: "0\\€", locale })).toBe("5€");
+    expect(() => formatValue(5, { format: "0€", locale })).toThrow();
+    expect(formatValue(5, { format: "0[$€]", locale })).toBe("5€");
+    expect(formatValue(5, { format: '0"€"', locale })).toBe("5€");
+    expect(formatValue(5, { format: '0"+"', locale })).toBe("5+");
+
+    expect(() => formatValue(5, { format: '0"€', locale })).toThrow();
+    expect(() => formatValue(5, { format: "0[$", locale })).toThrow();
+    expect(() => formatValue(5, { format: "0\\", locale })).toThrow();
+  });
+
+  test("Some characters are always escaped in format string", () => {
+    for (const char of "$+-/():!^&~{}<>= ") {
+      expect(formatValue(5, { format: `0${char}`, locale })).toBe(`5${char}`);
+    }
+  });
+
+  test("Escaped string in the middle of a number format", () => {
+    expect(formatValue(1234, { format: '0"Str"000', locale })).toBe("1Str234");
+    expect(formatValue(1234, { format: '#,##0"Str"0', locale })).toBe("1,23Str4");
+    expect(formatValue(1234, { format: '#,##0"Str"0.00', locale })).toBe("1,23Str4.00");
+    expect(formatValue(1.234, { format: '0.0"Str"0', locale })).toBe("1.2Str3");
+    expect(formatValue(1234, { format: '0.0"Str"0,', locale })).toBe("1.2Str3");
+  });
+
+  test("Multiple escaped strings in a number format", () => {
+    expect(formatValue(1234, { format: '[$$]0"€"', locale })).toBe("$1234€");
+    expect(formatValue(1234, { format: "[$$]0.00[$€]0", locale })).toBe("$1234.00€0");
+    expect(formatValue(1234, { format: '0"$"0"€"', locale })).toBe("123$4€");
   });
 
   test.each(["#,##0 dd", "dd #,##0", "#,##0 dd/mm/yyyy", "dd/mm/yyyy #,##0"])(
     "mixing date and numbers",
     (format) => {
       expect(() => formatValue(1234, { format, locale })).toThrow(
-        `Invalid number format: ${format}`
+        `Invalid first format part of: ${format}`
       );
```

Comment:
test that mixing date and number on different part of the format not throw:
ex `#,##0; dd`